### ISSUE TITLE
Execution stops on Provider.of call

### ISSUE THREAD

**Issue by emaborsa** [rrousselGit/provider#804]
Flutter 3.7.2 • channel stable • https://github.com/flutter/flutter.git
Framework • revision 32fb2f948e (5 days ago) • 2023-02-08 07:30:10 -0800
Engine • revision f40b73f8a4
Tools • Dart 2.19.2 • DevTools 2.20.1
Platform Android
Debugging on Huawei P20

I am working with Azure B2C, need to add Authentication, I found [this](https://pub.dev/packages/flutter_azure_b2c) library.
I have the following code:

```
Future<void> main() async {
  runApp(const App());
}

class App extends StatefulWidget {
  const App({super.key});

  @override
  AppState createState() => AppState();
}
Future<void> main() async {
  runApp(const App());
}

class App extends StatefulWidget {
  const App({super.key});

  @override
  AppState createState() => AppState();
}

class AppState extends State<App> {
  @override
  Widget build(BuildContext context) {
    return ChangeNotifierProvider(
      create: (context) => LoginModel(),
      child: MaterialApp(
        debugShowCheckedModeBanner: false,
        theme: appTheme,
        home: Consumer<LoginModel>(
          builder: (context, value, child) =>
              value.claims != null ? const Home() : const Login(),
        ),
      ),
    );
  }
}


class Login extends StatefulWidget {
  const Login({super.key});

  @override
  LoginState createState() => LoginState();
}

class LoginState extends State<Login> {
  B2CConfiguration? _configuration;

  checkLogin(BuildContext context) async {
    List<String>? subjects = await AzureB2C.getSubjects();

    if (subjects != null && subjects.isNotEmpty) {
      B2CAccessToken? token = await AzureB2C.getAccessToken(subjects[0]);
      if (!mounted || token == null) return;
      final encodedData = token.token.split('.')[1];
      final data = utf8.fuse(base64).decode(base64.normalize(encodedData));
      final claims = Claims.fromJson(jsonDecode(data));
      print("THIS IS PRINTED");
      var m = Provider.of<LoginModel>(context, listen: true);
      m.logIn(claims);
      print("THIS IS NEVER PRINTED");
    }
  }

  @override
  Widget build(BuildContext context) {
    AzureB2C.registerCallback(B2COperationSource.INIT, (result) async {
      if (result.reason == B2COperationState.SUCCESS) {
        _configuration = await AzureB2C.getConfiguration();
        if (!mounted) return;
        await checkLogin(context);
      }
    });

    AzureB2C.registerCallback(B2COperationSource.POLICY_TRIGGER_INTERACTIVE,
        (result) async {
      if (result.reason == B2COperationState.SUCCESS) {
        if (!mounted) return;
        await checkLogin(context);
      }
    });

    AzureB2C.handleRedirectFuture().then((_) => AzureB2C.init("auth_config"));

    return SafeArea(
      // omit code
    )
      
  }
}
```

Basically, on `TextButton.onPress` (omitted above) the B2C Login screen is opened, the user types username and password, if successful, the second callback is triggered. `THIS IS PRINTED` is printed, but `THIS IS NEVER PRINTED` is never. It seems that the execution of the code stops inside the call of `Provider.of`, I have no errors in the log.

Any idea?

**Comment by rrousselGit** [comment]
You probably have either an exception or an infinite loop.

In any case, not much I can do here. It's most certainly an issue on your side

**Comment by emaborsa** [comment]
Well, an infinite loop where? I don't have any...

**Comment by emaborsa** [comment]
I refactored the code, moved the `AzureB2C.registerCallback` into the initState method, seems to work.
Thanks.
